lib/OperatingSystems/DefaultOs.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @author Frank Karlitschek <frank@nextcloud.com>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Serverinfo\OperatingSystems;
+
+
+/**
+ * Class Ubuntu
+ *
+ * @package OCA\ServerInfo\OperatingSystems
+ */
+class DefaultOs {
+
+	/**
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function supported() {
+		return true;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getHostname() {
+		$hostname = shell_exec('hostname');
+		return $hostname;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getMemory() {
+		$memory = shell_exec('cat /proc/meminfo  | grep -i \'MemTotal\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		$memory = explode(' ',$memory);
+		$memory = round($memory[0]/1024);
+		if ($memory<1024) {
+			$memory = $memory.' MB';
+		} else {
+			$memory = round($memory/1024,1).' GB';
+		}
+		return $memory;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getCPUName() {
+		$cpu   = shell_exec('cat /proc/cpuinfo  | grep -i \'Model name\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		$cores = shell_exec('cat /proc/cpuinfo  | grep -i \'cpu cores\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		if ($cores == 1) $cores = ' ('.$cores.' core)'; else $cores = ' ('.$cores.' cores)';
+		return $cpu.' '.$cores;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getTime() {
+		$uptime = shell_exec('date');
+		return $uptime;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getUptime() {
+		$uptime = shell_exec('uptime -p');
+		return $uptime;
+	}
+
+
+	/**
+	* @return string
+	*/
+	public function getTimeServers() {
+		$uptime = shell_exec('cat /etc/ntp.conf |grep  \'^pool\' | cut -f 2 -d " "');
+		return $uptime;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getNetworkInfo() {
+			$result=array();
+			$result['hostname'] = \gethostname();
+			$dns = shell_exec('cat /etc/resolv.conf |grep -i \'^nameserver\'|head -n1|cut -d \' \' -f2');
```

Comment:
If I'm right this has also changed in Ubuntu 18.04 with systemd, where the integrated systemd-dns with entry `nameserver 127.0.0.53` is being used. But to be honest, I've no clue how to efficiently get the current DNS servers which are in use. According to [stackoverflow here](https://stackoverflow.com/questions/50299241/ubuntu-18-04-server-how-to-check-dns-ip-server-setting-being-used/50329033#50329033) we can get the current DNS server by parsing the output of `systemd-resolve --status`...

Reply:
yes. we need a different backend for this case then.